pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/StagePlanner.java

```diff
@@ -105,30 +113,128 @@ private StageNode walkRelPlan(RelNode node, int currentStageId) {
       RelDistribution.Type exchangeType = distribution.getType();
 
       // 2. make an exchange sender and receiver node pair
-      StageNode mailboxReceiver = new MailboxReceiveNode(currentStageId, nextStageRoot.getDataSchema(),
-          nextStageRoot.getStageId(), exchangeType);
-      StageNode mailboxSender = new MailboxSendNode(nextStageRoot.getStageId(), nextStageRoot.getDataSchema(),
-          mailboxReceiver.getStageId(), exchangeType, exchangeType == RelDistribution.Type.HASH_DISTRIBUTED
-          ? new FieldSelectionKeySelector(distributionKeys) : null);
+      KeySelector<Object[], Object[]> keySelector = exchangeType == RelDistribution.Type.HASH_DISTRIBUTED
+          ? new FieldSelectionKeySelector(distributionKeys) : null;
```

Comment:
Might want to add a comment here to explain the rationale. Something like this ?

Basically, we currently support two kinds of exchanges (BROADCAST and HASH_DISTRIBUTE) via hints. Only in the latter scenario as of now, there is a case for optimizing the exchange and considering skipping it. If the exchange type is BROADCAST, we won't have this situation because otherwise it becomes sort of colocated partitioned join (which I think will be added later and should have used a different exchange type through a different hint) .....

Reply:
good point. i will add the comments.